neutron-org/neutron
# feat: tx query result callback

This PR introduces a lot of changes:

1. The `SudoTxQueryResult()` callback is added. This callback is executed when a transaction result is submitted (unconditionally, every time);
2. Results for Tx queries **are not saved to state anymore**. Only the fact that a certain result (query_id + tx_hash) was submitted is saved, and duplicate results **are discarded without an error**;
3. Everything related to `QueryTransactions()` was removed.

**HOW TO TEST THIS CODE?**

Please have a look at this PR: https://github.com/neutron-org/neutron-contracts/pull/6
You can find the instructions in the readme: https://github.com/neutron-org/neutron-contracts/blob/feat/check-tx-query-result-callback/contracts/neutron_interchain_queries/README.md

**UPD:**

After a long discussion with @pr0n00gler we decided to limit the max number of transactions per submission to exactly 1.

Please re-review this PR, and please note that you now have to test it with this version of the relayer: https://github.com/neutron-org/cosmos-query-relayer/pull/16
